src/iso15118/io/connection_openssl.cpp

```diff
@@ -107,7 +107,7 @@ static SSL_CTX* init_ssl(const config::SSLConfig& ssl_config) {
                 return;
             }
             const auto result = key_logging_server.send(line);
-            if (result != strlen(line)) {
+            if (result < 0 or (result >= 0 and static_cast<size_t>(result) != strlen(line))) {
```

Comment:
`result < 0 or (static_cast<size_t>(result) != strlen(line))` should be enough I guess.  If this pattern is used far more often, we could also define a helper template like
```
template <typename Signed, typename Unsigned>
constexpr bool safe_compare(Signed signed_value, Unsigned unsigned_value) noexcept {
    static_assert(std::is_signed_v<Signed>, "signed parameter needs to be signed type");
    static_assert(std::is_unsigned_v<Unsigned>, "unsigned parameter needs to be unsigned type");

    if (signed_value < 0) {
        return false;
    }

    if (signed_value > std::numeric_limits<Unsigned>::max()) {
        return false;
    }

    return static_cast<Unsigned>(signed_value) == unsigned_value;
}
```
Probably it is not generic enough and has some bugs, but you get the idea

Reply:
who would have known:
https://en.cppreference.com/w/cpp/utility/intcmp
so this should be the way to go:
```
template<class T, class U>
constexpr bool cmp_equal(T t, U u) noexcept
{
    if constexpr (std::is_signed_v<T> == std::is_signed_v<U>)
        return t == u;
    else if constexpr (std::is_signed_v<T>)
        return t >= 0 && std::make_unsigned_t<T>(t) == u;
    else
        return u >= 0 && std::make_unsigned_t<U>(u) == t;
}
```

Reply:
Done, thx for finding `intcmp`!